File: android/src/main/java/net/xpece/android/text/NoUnderlineSpan.java
package net.xpece.android.text;

import android.text.TextPaint;
import android.text.style.CharacterStyle;
import android.text.style.UpdateAppearance;

/**
 * Removes underline e.g. from {@link android.text.style.URLSpan}.
 */

public class NoUnderlineSpan extends CharacterStyle implements UpdateAppearance {
    @Override
    public void updateDrawState(final TextPaint tp) {
        tp.setUnderlineText(false);
    }
}

File: text-span/src/main/java/net/xpece/android/text/span/NoUnderlineSpan.kt
package net.xpece.android.text.span

import android.text.TextPaint
import android.text.style.CharacterStyle
import android.text.style.UpdateAppearance

/**
 * Removes underline e.g. from [android.text.style.URLSpan].
 */
class NoUnderlineSpan : CharacterStyle(), UpdateAppearance {

    override fun updateDrawState(tp: TextPaint) {
        tp.isUnderlineText = false
    }
}
